libraries/wallet/wallet.cpp

```diff
@@ -758,6 +758,56 @@ class wallet_api_impl
 
       return true;
    }
+
+   signed_transaction multisig_sign_transaction( signed_transaction tx,
+                                                 const vector<string> &wif_keys, bool broadcast )
+   {
+      vector<fc::ecc::private_key> private_keys;
+
+      for ( const string &wif_key : wif_keys )
+      {
+         fc::optional<fc::ecc::private_key> optional_private_key = wif_to_key( wif_key );
+
+         if ( !optional_private_key )
+         {
+            elog( "Invalid private key: ${i}", ( "i", wif_key ) );
+            FC_THROW("");
+         }
+
+         private_keys.push_back( *optional_private_key );
+      }
+
+      for ( const fc::ecc::private_key &privkey : private_keys )
+      {
+         signature_type sig = privkey.sign_compact( tx.sig_digest( _chain_id ) );
+         const vector<signature_type> &sigs = tx.signatures;
+         if ( std::find( sigs.begin(), sigs.end(), sig ) != sigs.end() )
```

Comment:
That doesn't work. Calling sign_compact twice will produce different signatures.
If you want to find out if a given key has already signed a tx, you must derive the public keys from the existing signatures and compare those. See sign_state in libraries/chain/protocol/transaction.cpp.

Reply:
Sorry not understand here.. did you mean the 2nd call of `sign_compact` will compute the signature of the transaction plus 1st computed signature, not the transaction itself?

I have a test case in `tests/cli/main.cpp`, shows that when signing the transaction twice with one private key, the transaction resulted in only one signature.

Reply:
No, I meant that the signature returned by privkey.sign_compact will never be found in the sigs list, because every call to sign_compact will create a new, unique signature.

Actually it turns out that privkey.sign_compact uses a deterministic signature generation scheme, which means that the signatures created by your test case are in fact found in the sigs list, so the test works as expected. However, there is no guarantee that this will always be the case - generally, it is possible that a given private key produces different signatures for a given tx.